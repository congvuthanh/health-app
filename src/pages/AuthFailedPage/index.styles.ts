import { Link } from 'react-router';
import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  min-height: 100vh;
  padding: 2rem;
  background-color: ${({ theme }) => theme.colors.white};

  @media (max-width: ${({ theme }) => theme.breakpoints.mobile}) {
    padding: 1.6rem;
  }
`;

export const Content = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  text-align: center;
  max-width: 60rem;
  width: 100%;
`;

export const Title = styled.h1`
  ${({ theme }) => theme.typography.h3.bold}
  font-family: ${({ theme }) => theme.fonts.japanese};
  color: ${({ theme }) => theme.colors.primary[500]};
  margin-bottom: 2.4rem;

  @media (max-width: ${({ theme }) => theme.breakpoints.mobile}) {
    ${({ theme }) => theme.typography.h4.bold}
    margin-bottom: 1.6rem;
  }
`;

export const Description = styled.p`
  ${({ theme }) => theme.typography.p2.regular}
  font-family: ${({ theme }) => theme.fonts.japanese};
  color: ${({ theme }) => theme.colors.black[600]};

  @media (max-width: ${({ theme }) => theme.breakpoints.mobile}) {
    ${({ theme }) => theme.typography.p3.regular}
  }
`;

export const LoginLink = styled(Link)`
  ${({ theme }) => theme.typography.p2.bold}
  font-family: ${({ theme }) => theme.fonts.japanese};
  color: ${({ theme }) => theme.colors.primary[500]};
  text-decoration: underline;
  margin-left: 0.4rem;
  transition: opacity 0.2s ease;

  &:hover {
    opacity: 0.8;
  }

  &:focus {
    outline: 2px solid ${({ theme }) => theme.colors.primary[500]};
    outline-offset: 2px;
  }

  @media (max-width: ${({ theme }) => theme.breakpoints.mobile}) {
    ${({ theme }) => theme.typography.p3.bold}
  }
`;
