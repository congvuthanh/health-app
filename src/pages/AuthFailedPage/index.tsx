import styled from 'styled-components';

const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  min-height: 100vh;
  padding: 2rem;
`;

const Title = styled.h1`
  font-size: 2rem;
  font-weight: 700;
  margin-bottom: 1rem;
  color: #e74c3c;
`;

function AuthFailedPage() {
  return (
    <Container>
      <Title>Authentication Failed</Title>
      <p>Authentication Error - URL path: /authenticationError</p>
    </Container>
  );
}

export default AuthFailedPage;
