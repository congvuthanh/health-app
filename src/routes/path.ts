export const routePath = {
  TopPage: '/',
  MyPage: '/myPage',
  AuthFailedPage: '/authenticationError',
} as const;

type ValueOf<T> = T[keyof T];
type P = ValueOf<typeof routePath>;

export type AppPath = P extends infer T
  ? T extends (...subPath: string[]) => infer U
    ? U
    : T
  : never;
