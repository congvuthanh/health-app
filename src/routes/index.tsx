import { createBrowserRouter, RouterProvider } from 'react-router';
import AuthFailedPage from '../pages/AuthFailedPage';
import MyPage from '../pages/MyPage';
import NotFoundPage from '../pages/NotFoundPage';
import TopPage from '../pages/TopPage';

const router = createBrowserRouter([
  {
    path: '/',
    element: <TopPage />,
  },
  {
    path: '/myPage',
    element: <MyPage />,
  },
  {
    path: '/authenticationError',
    element: <AuthFailedPage />,
  },
  {
    path: '*',
    element: <NotFoundPage />,
  },
]);

export function AppRouter() {
  return <RouterProvider router={router} />;
}
