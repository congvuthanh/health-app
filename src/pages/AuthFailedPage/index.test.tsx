import { screen } from '@testing-library/react';
import { routePath } from 'routes/path';
import { renderWithProviders } from 'utils/test/test-utils';
import { describe, expect, it } from 'vitest';
import AuthFailedPage from './index';

describe('AuthFailedPage', () => {
  it('renders the authentication error page', () => {
    renderWithProviders(<AuthFailedPage />);

    expect(screen.getByText('ログインが必要です')).toBeInTheDocument();
  });

  it('displays the description text', () => {
    renderWithProviders(<AuthFailedPage />);

    expect(screen.getByText(/ログインページは/)).toBeInTheDocument();
    expect(screen.getByText('こちら')).toBeInTheDocument();
  });

  it('renders a link to the login page', () => {
    renderWithProviders(<AuthFailedPage />);

    const loginLink = screen.getByRole('link', { name: 'こちら' });
    expect(loginLink).toBeInTheDocument();
    expect(loginLink).toHaveAttribute('href', routePath.TopPage);
  });

  it('has proper accessibility attributes for the link', () => {
    renderWithProviders(<AuthFailedPage />);

    const loginLink = screen.getByRole('link', { name: 'こちら' });
    expect(loginLink).toBeVisible();
  });

  it('renders the title with correct heading level', () => {
    renderWithProviders(<AuthFailedPage />);

    const title = screen.getByRole('heading', { name: 'ログインが必要です' });
    expect(title).toBeInTheDocument();
    expect(title.tagName).toBe('H1');
  });

  it('matches snapshot', () => {
    const { container } = renderWithProviders(<AuthFailedPage />);
    expect(container).toMatchSnapshot();
  });
});
