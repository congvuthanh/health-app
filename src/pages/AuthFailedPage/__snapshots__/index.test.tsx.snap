// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AuthFailedPage > matches snapshot 1`] = `
<div>
  <div
    class="sc-bRKDuR jtrmVE"
  >
    <div
      class="sc-hvigdm iMgZNU"
    >
      <h1
        class="sc-fhHczv jA-dCcq"
      >
        ログインが必要です
      </h1>
      <p
        class="sc-ggWZvA bhbFQv"
      >
        ログインページは
        <a
          class="sc-dTvVRJ iudMkS"
          data-discover="true"
          href="/"
        >
          こちら
        </a>
      </p>
    </div>
  </div>
</div>
`;
